### ISSUE TITLE
SA1119: StatementMustNotUseUnnecessaryParenthesis

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#88]
http://www.stylecop.com/docs/SA1119.html

**Comment by pdelvo** [comment]
grabbed. This is going to be a very annoying one to write all the tests for.

**Comment by pdelvo** [comment]
StyleCop's implementation of this seems very weird.
Does not report SA1119:
```csharp
test = ((x) + "abc").ToString();
```
Does report SA1119:
```csharp
test = ((x).ToString() + "abc").ToString();
```
It also does not report anything for
```csharp
Func<string, object> f = x => (x = "t");
```
Should I follow the StyleCop way or do it differently?

It is also fine with parenthesis in binary expressions even if they are not needed. Example:
```csharp
int x = (1 + 1) + 1;
```
Any thoughts on that if this should stay like this?

**Comment by sharwell** [comment]
I have no idea. I recommend testing this on a large code base and see if the warnings it produces are sensible.

Seems like in general we would want this to report less items rather than more.
